src/amo/reducers/search.js

```diff
@@ -125,6 +129,7 @@ export default function search(
         results: payload.results.map((addon) =>
           createInternalAddon(addon, state.lang),
         ),
+        pageCount: payload.pageCount,
```

Comment:
In addition to this, we should probably set `pageCount` to `0` (or maybe to `null`, but `0` is probably fine) in both `SEARCH_STARTED` and `SEARCH_ABORTED` so that it gets reset each time a search is run, and doesn't carry over the value from the last search.